### DISCUSSION TITLE
Abstracts in Publication Objects

### DISCUSSION THREAD

**Discussion by josephdicastro** [scholarly-python-package/scholarly#398] [Q&A] [answered]
Hi! Thanks for your great package! 

I'm a little confused as to what data is appearing in the abstract container in a Publication object. Is it just the snippet from the organic search results? Or is a complete abstract also available (based on a complete one even being available in Google Scholar, of course) 

Thanks!

**Comment by ipeirotis** [comment, marked as answer]
Depends on the source.

If the publication source is the search results, it will be whatever snippet Google shows under the search result.

If the publication source is the author page, then the abstract will typically be the full abstract of the paper.

**Reply by josephdicastro** [reply]
Thank you! 

Is the source you mention dependent on which object in used in Scholarly to query with, authors vs publications? Or are you saying that it depends on Google's index where they are extracting the data from?

**Reply by ipeirotis** [reply]
When you use the `search_pubs` the publications come from the main search interface, so abstracts are snippets.

If you do a `search_author` and get the publications from the author, then the paper entries will typically be more complete.

**Reply by josephdicastro** [reply]
Thank you so much!
